homeassistant/helpers/config_entry_oauth2_flow.py

```diff
@@ -259,10 +259,21 @@ def extra_authorize_data(self) -> dict:
         """
         return self.async_create_entry(title=self.flow_impl.name, data=data)
 
-    async_step_user = async_step_pick_implementation
-    async_step_ssdp = async_step_pick_implementation
-    async_step_zeroconf = async_step_pick_implementation
-    async_step_homekit = async_step_pick_implementation
+    async def async_step_discovery(self, user_input: dict = None) -> dict:
+        """Handle a flow initialized by discovery."""
+        await self.async_set_unique_id(self.DOMAIN)
+        self._abort_if_unique_id_configured()
+
+        assert self.hass is not None
```

Comment:
Why do this assertion? Assert should normally not be used in production code.

Reply:
mypy 🙄

Reply:
I made a design mistake when making the config flow and make it so that the parent sets the `hass` instead of passing it in the constructor. Now `mypy` can't know that `hass` is set.